src/lib/dom.generated.d.ts

```diff
@@ -2292,6 +2362,39 @@ interface ULongRange {
     min?: number;
 }
 
+interface URLPatternComponentResult {
+    groups?: Record<string, string | undefined>;
+    input?: string;
+}
+
+interface URLPatternInit {
+    baseURL?: string;
+    hash?: string;
+    hostname?: string;
+    password?: string;
+    pathname?: string;
+    port?: string;
+    protocol?: string;
+    search?: string;
+    username?: string;
+}
+
+interface URLPatternOptions {
+    ignoreCase?: boolean;
+}
+
+interface URLPatternResult {
+    hash?: URLPatternComponentResult;
+    hostname?: URLPatternComponentResult;
+    inputs?: URLPatternInput[];
+    password?: URLPatternComponentResult;
+    pathname?: URLPatternComponentResult;
+    port?: URLPatternComponentResult;
+    protocol?: URLPatternComponentResult;
+    search?: URLPatternComponentResult;
+    username?: URLPatternComponentResult;
+}
```

Comment:
Perhaps I'm missing something, but Node defines these types without `undefined` and https://urlpattern.spec.whatwg.org/#dictdef-urlpatternresult sort of implies they are always defined. Is that incorrect? (I am not super familiar with reading these specs).

Or is it that `dictionary` means all props can be undefined?

@saschanaz 😅